openmls/src/treesync/mod.rs

```diff
@@ -441,24 +463,52 @@ impl TreeSync {
     ///
     /// This function should not fail and only returns a [`Result`], because it
     /// might throw a [LibraryError](TreeSyncError::LibraryError).
-    pub(crate) fn own_leaf_node(&self) -> Result<&LeafNode, TreeSyncError> {
+    pub(crate) fn own_leaf_node(&self) -> Result<&OpenMlsLeafNode, TreeSyncError> {
         // Our own leaf should be inside of the tree and never blank.
         self.leaf(self.own_leaf_index)?
             .ok_or_else(|| LibraryError::custom("Own leaf is outside of the tree").into())
     }
 
+    /// Returns the [`LeafNode`] of this client.
+    ///
+    /// This function should not fail and only returns a [`Result`], because it
+    /// might throw a [LibraryError](TreeSyncError::LibraryError).
+    pub(crate) fn own_leaf_node_mut(&mut self) -> Result<&mut OpenMlsLeafNode, TreeSyncError> {
```

Comment:
This kind of breaks the pattern where all changes to the TreeSync instance have to happen by merging a diff.